packages/slice-machine/components/Tooltip/TextWithTooltip.tsx

```diff
@@ -1,38 +1,54 @@
 import ReactTooltip from "react-tooltip";
-import React, { useEffect, useRef, useState } from "react";
-import { Heading } from "theme-ui";
+import React, { ElementType, useEffect, useRef, useState } from "react";
+import { Text, ThemeUIStyleObject } from "theme-ui";
 
-interface HeadingWithTooltip {
+interface TextWithTooltipProps {
   text: string;
+  as: ElementType;
+  sx?: ThemeUIStyleObject;
 }
 
-export const HeadingWithTooltip: React.FC<HeadingWithTooltip> = ({ text }) => {
+export const TextWithTooltip: React.FC<TextWithTooltipProps> = ({
+  text,
+  as,
+  sx = {},
+}) => {
   const headingRef = useRef<HTMLDivElement>(null);
   const [showTooltip, setShowTooltip] = useState<boolean | null>(null);
 
-  useEffect(() => {
+  const updateTooltipVisibility = () => {
     setShowTooltip(
       headingRef.current &&
         headingRef.current.offsetWidth < headingRef.current.scrollWidth
     );
+  };
+
+  useEffect(() => {
+    updateTooltipVisibility();
+    window.addEventListener("resize", updateTooltipVisibility);
```

Comment:
cool! So this triggers the updateTooltipVisibility when the window gets resized?

Reply:
yes, took a bit to realise why it wasn't working lol, the showTooltip was only computed once at first render and never updated